pysr/regressor_sequence.py

```diff
@@ -0,0 +1,299 @@
+                    for j in range(self.recursive_history_length, 0, -1)
+                    for i in range(n_features)
+                ]
+        else:
+            return [
+                i + "t_" + str(j)
+                for j in range(self.recursive_history_length, 0, -1)
+                for i in variable_names
+            ]
+
+    def fit(
+        self,
+        X,
+        *,
+        weights=None,
+        variable_names: Optional[List[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+    ) -> "PySRSequenceRegressor":
+        """
+        Search for equations to fit the sequence and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Sequence of shape (n_times, n_features).
+        weights : ndarray | pandas.DataFrame
+            Weight array of the same shape as `X`.
+            Each element is how to weight the mean-square-error loss
+            for that particular element of `X`. Alternatively,
+            if a custom `loss` was set, it will can be used
+            in arbitrary ways.
+        variable_names : list[str]
+            A list of names for the variables, rather than "x0t_1", "x1t_2", etc.
+            If `X` is a pandas dataframe, the column name will be used
+            instead of `variable_names`. Cannot contain spaces or special
+            characters. Avoid variable names which are also
+            function names in `sympy`, such as "N".
+            The number of variable names must be equal to (n_features,).
+        complexity_of_variables : int | float | list[int] | list[float]
+            The complexity of each variable in `X`. If a single value is
+            passed, it will be used for all variables. If a list is passed,
+            its length must be the same as recurrence_history_length.
+        X_units : list[str]
+            A list of units for each variable in `X`. Each unit should be
+            a string representing a Julia expression. See DynamicQuantities.jl
+            https://symbolicml.org/DynamicQuantities.jl/dev/units/ for more
+            information.
+            Length should be equal to n_features.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+        X = self._validate_data(X)
+        _check_assertions(
+            X,
+            self.recursive_history_length,
+            weights,
+            variable_names,
+            X_units,
+        )
+        self.variable_names = variable_names  # for latex_table()
+        self.n_features = X.shape[1]  # for latex_table()
+
+        current_X = X[self.recursive_history_length :]
+        historical_X = np.lib.stride_tricks.sliding_window_view(
+            X[:-1].flatten(), self.recursive_history_length * X.shape[1]
+        )[:: current_X.shape[1], :]
```

Comment:
Since this code is similar to the one used in `predict`, could we move it to a separate internal function? That way there is only one appearance of `np.lib.stride_tricks.sliding_window_view` and the name of that internal function would make its behavior clear.

Reply:
unfortunately in predict() there is no [:-1] but there is a [:-1] in fit()
I'll add a slide_up_to parameter tho?

Reply:
actually predict() has [:: X.shape[1] :] but fit() has [:: current_X.shape[1] :]
I'll probably just move the sliding_window_view to another function without the slice